papyri/gen.py

```diff
@@ -550,6 +551,8 @@ def gen_main(
 
     if not target_dir.exists() and not config.dry_run:
         target_dir.mkdir(parents=True, exist_ok=True)
+    if dry_run:
+        target_dir = tempfile.mkdtemp()
```

Comment:
And the linter mypy in unhappy as targetdir is supposed to be a `Path`, you you at least want to make

```
target_dir = Path(tempfile.mkdtemp())
```